R/db-interface.R

```diff
@@ -190,7 +190,10 @@ copy_dm_to <- function(dest, dm, ...,
   # FIXME: if same_src(), can use compute() but need to set NOT NULL and other
   # constraints
 
-  dm <- collect(dm, progress = progress)
+  # use 0-rows dm object from now on
+  dmdata <- dm ## we still need it to copy actual data, we will collect tables one by one to reduce peak memory usage
+  dm <- dm |> dm_ptype() |> collect()
+  #dm <- collect(dm, progress = progress)
```

Comment:
We can't use `|>` yet, for compatibility. Seems cleaner to introduce a new variable name (and perhaps keep `dm` unchanged).

```suggestion
  ptype_dm <- collect(dm_ptype(dm))
```